src/libcore/pin.rs

```diff
@@ -584,6 +584,25 @@ impl<P: DerefMut> Pin<P> {
     /// the pointee cannot move after `Pin<Pointer<T>>` got created.
     /// "Malicious" implementations of `Pointer::DerefMut` are likewise
     /// ruled out by the contract of `Pin::new_unchecked`.
+    ///
+    /// This method is useful when calling multiple times a function that consumes the pinned type.
```

Comment:
```suggestion
    /// This method is useful when calling a function that consumes the pinned type multiple times.
```

Reply:
This now reads like "a function that consumes the pinned type multiple times", but how can a function consume an argument multiple times?

What about "when doing multiple calls to functions that consume the pinned type"?

Reply:
> What about "when doing multiple calls to functions that consume the pinned type"?

Seems to me better.